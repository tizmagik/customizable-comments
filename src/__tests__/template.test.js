const prOpened = require('./fixtures/pull_request.opened.json');

jest.setMock('probot-config', () => {
  return Promise.resolve({
    vars: [
      {
        name: '$BRANCH_SUFFIX',
        value: '(test-suffix)'
      }
    ],
    pull_request: {
      opened: {
        template: `
    Here's a helpful URL based on the branch name: https://$BRANCH-$BRANCH_SUFFIX.something.example.com
    Second line goes here. 
    BRANCH is: $BRANCH
    BRANCH_SUFFIX is: $BRANCH_SUFFIX
    BRANCH_SANITIZED is: $BRANCH_SANITIZED
    Done.
        `
      }
    }
  });
});

describe('template', () => {
  const { template } = require('../template');
  it('stuff', async done => {
    const compiled = await template(prOpened, 'pull_request.opened');
    expect(compiled).toMatchSnapshot();
    done();
  });
});
