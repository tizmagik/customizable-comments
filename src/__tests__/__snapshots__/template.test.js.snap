// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template stuff 1`] = `
"
    Here's a helpful URL based on the branch name: https://feature/PRJ-some.branch/test-123-(test-suffix).something.example.com
    Second line goes here. 
    BRANCH is: feature/PRJ-some.branch/test-123
    BRANCH_SUFFIX is: (test-suffix)
    BRANCH_SANITIZED is: feature-prj-some-branch-test-123
    Done.
        "
`;
